src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsMigrationService.kt

```diff
@@ -65,9 +66,22 @@ class NonAssociationsMigrationService(
   }
 
   override suspend fun migrateNomisEntity(context: MigrationContext<NonAssociationIdResponse>) {
-    log.info("attempting to migrate ${context.body}")
-    val firstOffenderNo = context.body.offenderNo1
-    val secondOffenderNo = context.body.offenderNo2
+    val nonAssociation = context.body
+    log.info("attempting to migrate $nonAssociation")
+    val firstOffenderNo = nonAssociation.offenderNo1
+    val secondOffenderNo = nonAssociation.offenderNo2
+
+    if (nonAssociation.isNotPrimaryNonAssociation()) {
+      telemetryClient.trackEvent(
+        "non-association-migration-non-primary-skipped",
+        mapOf(
+          "firstOffenderNo" to firstOffenderNo,
+          "secondOffenderNo" to secondOffenderNo,
+          "migrationId" to context.migrationId,
+        ),
+      )
+      return
```

Comment:
`return` not doing anything?

Reply:
return just bombs out - it follows the pattern done in other service methods where we don't process the event if we're not interested

Reply:
sorry, I hadn't realised there was more code after the return 🙁